framework/include/base/ComputeNodalKernelBCJacobiansThread.h

```diff
@@ -0,0 +1,53 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef COMPUTENODALKERNELBCJACOBIANSTHREAD_H
+#define COMPUTENODALKERNELBCJACOBIANSTHREAD_H
+
+#include "ParallelUniqueId.h"
+#include "NodalKernelWarehouse.h"
+#include "MooseMesh.h"
+
+// libMesh includes
+#include "libmesh/node_range.h"
+#include "libmesh/numeric_vector.h"
+
+
+class FEProblem;
+class AuxiliarySystem;
+
+
+class ComputeNodalKernelBCJacobiansThread
```

Comment:
Should not this class inherit from `ThreadedNodeLoop<ConstNodeRange, ConstNodeRange::const_iterator>` ?